Fix/internal modifier parameter

Gabriel-Moraes350 opened the pull request:
This PR simulates and tries to fix the build issue due to the following scenario:
- Have a public class with a private method that is going to use the KTestable annotation
- A parameter of this private method is a internal or protected class
- The generated function has a public visibility modifier that triggers an error due to exposing a internal type

A example was created in the kotlin-gradle-kotlin-dsl-example.

ericjesse commented on catadioptre-annotations/src/main/kotlin/io/aerisconsulting/catadioptre/kotlin/KModifierClassification.kt:
```diff
@@ -0,0 +1,15 @@
+
+import com.squareup.kotlinpoet.KModifier
+
+enum class KModifierClassification(val kModifier: KModifier, val classification: Int) {
```
```suggestion
internal enum class KlassVisibility(val klassModifier: KModifier) {
```

ericjesse commented on catadioptre-annotations/src/main/kotlin/io/aerisconsulting/catadioptre/kotlin/KotlinTestableProcessor.kt:
```diff
@@ -125,6 +121,15 @@ internal class KotlinTestableProcessor : AbstractProcessor() {
         return true
     }
 
+    private fun getVisibilityModifier(classContainer: ImmutableKmWithFlags): KModifier {
```
```suggestion
    private fun getVisibilityModifier(classMetadata: ImmutableKmWithFlags): KlassVisibility {
```

ericjesse commented on catadioptre-annotations/src/main/kotlin/io/aerisconsulting/catadioptre/kotlin/KotlinTestableProcessor.kt:
```diff
@@ -277,4 +286,10 @@ internal class KotlinTestableProcessor : AbstractProcessor() {
 
+    private fun getLowestVisibility(parametersModifiers: MutableList<KModifier>, classVisibilityModifier: KModifier): KModifier {
+
+        val kModifierWithClassification = parametersModifiers.plus(classVisibilityModifier).map { KModifierClassification.getByKModifier(it) }
```
(parametersModifiers + classVisibilityModifier).minBy{ it.ordinal } should be shorter as a unique statement of the method.

Gabriel-Moraes350 commented on catadioptre-annotations/src/main/kotlin/io/aerisconsulting/catadioptre/kotlin/KotlinTestableProcessor.kt:
```diff
@@ -125,6 +121,15 @@ internal class KotlinTestableProcessor : AbstractProcessor() {
 
+    private fun getVisibilityModifier(classContainer: ImmutableKmWithFlags): KModifier {
+        return when {
+            classContainer.isPublic -> KModifier.PUBLIC
```
You're right, I didn't think about that, I'm changing it

Gabriel-Moraes350 replied:
I've debugged and created a test for it, and it seems that the change won't be needed. Or at least I was not able to simulate the scenario

Gabriel-Moraes350 commented:
I've updated the PR with the fixes.
I added a couple of tests for the Java generated code, just to verify that the visibility behaviour does not affect Java.
As we don't have the internal modifier in Java, I changed the test a little bit.
